### PR TITLE
Fix #8 - Support inverting logic within the filter

### LINKED ISSUE #8: support bloom filters that have more `include` items than `exclude`
If `len(include) > len(exclude)` then then an error is thrown:

```
/deps/lib/python3.6/site-packages/filtercascade/__init__.py:268: in cascade_with_characteristics
    [Bloomer.filter_with_characteristics(capacity, error_rates[0])],
/deps/lib/python3.6/site-packages/filtercascade/__init__.py:75: in filter_with_characteristics
    return Bloomer(size=size, nHashFuncs=nHashFuncs, level=level)
/deps/lib/python3.6/site-packages/filtercascade/__init__.py:30: in __init__
    self.bitarray = bitarray.bitarray(self.size, endian='little')
E   ValueError: cannot create bitarray with negative length
```

For the addon blocklist it's possible that we will have at some point more blocked addons (include) than not blocked.

### PR DESCRIPTION
**Pull request by jcjones**
This adds the ability to logically invert the state of a filter cascade in the event that the exclude set is actually smaller than the input set. This also adds diagnostics to assist the user in selecting this, and attempts to set the inversion-bit automatically if the code can determine the correct answer before running.

### CODE REVIEW

File: filtercascade/__init__.py [hunk 1]
```diff
@@ -222,6 +252,15 @@ def initialize(self, *, include, exclude):
         except TypeError as te:
             raise TypeError("include is not a list", te)
 
+        if self.invertedLogic is None:
+            try:
+                self.invertedLogic = len(include) > len(exclude)
```
Comment (mozkeeler) [inline filtercascade/__init__.py:257]: Is the comment about `exclude` being larger than main memory still relevant? Does `len(exclude)` do something smart to not run out of memory in that case?

**Review by mozkeeler — APPROVED** [review]
Looks good to me - just the one comment.

File: filtercascade/__init__.py (hunk 1 above)
Comment (jcjones) [reply filtercascade/__init__.py:257]: If `exclude` is an iterator then `len(exclude)` will raise `TypeError` and we won't auto-set `self.invertedLogic`.  Good catch! :)

**Comment by jcjones** [comment]
Thanks, Dana!

**Pull request merged**